frontend/src/hooks/useEventsSearch.ts

```diff
@@ -0,0 +1,70 @@
+import { useMemo } from "react";
+import { TraceEvent } from "@/validators/TraceEvent";
+
+interface UseEventsSearchArguments {
+    searchValue: string;
+    searchBy: EventSearchCriteria;
+    data: TraceEvent[];
+}
+
+export type EventSearchCriteria = 'eventName' | 'eventDetails' | 'all';
+
+
+function getKeysForSearchedValue(obj: any, searchWord: string, parent = '', res: string[] = []) {
+    const lowerSearchWord = searchWord.toLowerCase();
+
+    if (typeof obj === 'object' && obj !== null) {
+        for (let key in obj) {
+            const propName = parent ? `${parent}.${key}` : key;
+
+            if (key.toLowerCase().includes(lowerSearchWord)) {
+                res.push(propName);
+            }
+
+            getKeysForSearchedValue(obj[key], lowerSearchWord, propName, res);
+        }
+    } else if (typeof obj === 'string' && obj.toLowerCase().includes(lowerSearchWord)) {
+        res.push(parent);
+    }
+
+    const result = res.flatMap(item => item.split('.'));
+
+    return result;
+}
+
+
+export const useEventsSearch = ({ searchValue, data, searchBy }: UseEventsSearchArguments) => {
+    const searchResult = useMemo(() => {
+        let keysNotToCollapse: string[] = ["root"];
```

Comment:
keysNotToCollpase need to be assigned to a specific event (selected event really).
Otherwise you get this bug:

https://github.com/user-attachments/assets/0a518f0b-5b81-4469-8579-03fe44b84a27